.github/workflows/CI.yml

```diff
@@ -0,0 +1,95 @@
+on:
+  push:
+    branches-ignore:
+      - 'staging.tmp'
+      - 'trying.tmp'
+
+name: CI
+
+jobs:
+  static_analysis:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v1
+
+      - name: Extract toolchain version from rust-toolchain
+        run: echo "::set-env name=RUST_TOOLCHAIN::$(cat rust-toolchain)"
+
+      - name: Install ${{ env.RUST_TOOLCHAIN }} toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          override: true
+
+      - name: Check formatting
+        run: make check_format
+
+      - name: Run linter
+        run: make clippy
+
+  build:
+    strategy:
+      matrix:
+        os: [macos, windows, ubuntu]
+        include:
+          - os: windows
+            binary: create-comit-app.exe
+          - os: ubuntu
+            binary: create-comit-app
+          - os: macos
+            binary: create-comit-app
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v1
+
+      - name: Extract toolchain version from rust-toolchain
+        run: echo "::set-env name=RUST_TOOLCHAIN::$(cat rust-toolchain)"
+
+      - name: Install ${{ env.RUST_TOOLCHAIN }} toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ env.RUST_TOOLCHAIN }}
+          override: true
+
+      - name: Build ${{ matrix.os }} binary
+        run: make build
+
+      - name: Run unit tests
+        run: make test
+
+      - name: Upload ${{ matrix.os }} binary
+        uses: actions/upload-artifact@v1
+        with:
+          name: create-comit-app-${{ matrix.os }}-debug
+          path: target/debug/${{ matrix.binary }}
+
+  e2e_test:
+    strategy:
+      matrix:
```

Comment:
:wrench: I guess you don't need a matrix if you only run it on 1 :stuck_out_tongue: